components/modbus_controller.rst

```diff
@@ -166,135 +161,262 @@ Some devices use decimal values in read registers to show multiple binary states
+      #   register_type: "read"
+      custom_command: [0x2, 0x4, 0x1, 0x58, 0x00, 0x02]
+      ## the command returns an float value using 4 bytes
+      lambda: |-
+          ESP_LOGD("Modbus Sensor Lambda","Got new data" );
+          union {
+            float float_value;
+            uint32_t raw;
+          } raw_to_float;
+          if (data.size() < 4 ) {
+            ESP_LOGE("Modbus Sensor Lambda", "invalid data size %d",data.size());
+            return NAN;
+          }
+          raw_to_float.raw =   data[0] << 24 | data[1] << 16 | data[2] << 8 |  data[3];
+          ESP_LOGD("Modbus Sensor Lambda", "FP32 = 0x%08X => %f", raw_to_float.raw, raw_to_float.float_value);
+          return raw_to_float.float_value;
+        unit_of_measurement: kVArh
+        accuracy_decimals: 1
+
+.. _modbus_register_count:
+
+Optimizing modbus communications
+--------------------------------
+
+``register_count`` is an option only required for uncommon response encodings or to optimizie modbus communications.
+
+It describes the number of registers this data point spans. Overrides the defaults determined by ``value_type``. If no value for ``register_count`` is provided, it is calculated based on the register type. The default size for 1 register is 16 bits (1 Word). Some devices are not adhering to this convention and have registers larger than 16 bits. In this case ``register_count`` and  ``response_size`` must be set. For example, if your modbus device uses 1 registers for a FP32 value instead the default of two set ``register_count: 1`` and ``response_size: 4``.
+
+``register_count`` can also be used to skip a number of registers in consecutive range.
+
+An example is an SDM meter, with interesting data in register addresses 0, 2, 4 and 6:
+
+.. code-block:: yaml
+
+    - platform: modbus_controller
+      name: "Voltage Phase 1"
+      address: 0
+      register_type: "read"
+      value_type: FP32
+
+    - platform: modbus_controller
+      name: "Voltage Phase 2"
+      address: 2
+      register_type: "read"
+      value_type: FP32
+
+    - platform: modbus_controller
+      name: "Voltage Phase 3"
+      address: 4
+      register_type: "read"
+      value_type: FP32
+
+    - platform: modbus_controller
+      name: "Current Phase 1"
+      address: 6
+      register_type: "read"
+      value_type: FP32
+      accuracy_decimals: 1
+
+The configuration above will generate *one* modbus command *read multiple registers from 0 to 6*.
+
+Maybe you don’t care about the data in register addresses 2 and 4, which are voltage values for Phase 2 and Phase 3 (or you have a SDM-120).
+Of course, you can delete the sensors your don’t care about. But then you have a gap in the addresses. If you remove the registers at address 2 and 4 then *two* commands will be generated *read register 0* and *read register 6*. To avoid the generation of multiple commands and reduce the amount of uart communication ``register_count`` can be used to fill the gaps:
```

Comment:
```suggestion
Of course, you can delete the sensors your don’t care about, but then you'd have a gap in the addresses. If you remove the registers at address 2 and 4, *two* commands will be generated -- *read register 0* and *read register 6*. To avoid generating multiple commands and thus reduce activity on the bus, ``register_count`` can be used to fill the gaps:
```